x/programs/runtime/runtime.go

```diff
@@ -72,6 +72,22 @@ func (r *WasmRuntime) AddProgram(program codec.Address, bytes []byte) (*Program,
 	return programModule, nil
 }
 
+func (r *WasmRuntime) CallProgramToResponse(ctx context.Context, callInfo *CallInfo) []byte {
+	result, err := r.CallProgram(ctx, callInfo)
+	if err != nil {
+		trap := convertToTrap(err)
+		switch *trap.Code() {
```

Comment:
I'm not sure this is safe, I don't think you are guaranteed to have a trap code

Reply:
And since the new once won't be matched with either of the unreachable or out of fuel, it would just default to `ExecutionFailure`

Reply:
I have done a conversion trial with `errors.As`. If that doesn't match as a trap error, then we just serialize to a `ExecutionFailure`.